augraphy/augmentations/markup.py

```diff
@@ -118,35 +127,47 @@ def __call__(self, image, layer=None, force=False):
-                        # A filled rectangle
-                        markup_mask = cv2.rectangle(markup_mask, p1, p2, self.markup_color, -1)
-                    else:
-                        p2 = (int(points_list[i + 1][0]), int(points_list[i + 1][1]))
-                        markup_mask = cv2.line(
-                            markup_mask,
-                            p1,
-                            p2,
-                            self.markup_color,
-                            markup_thickness,
-                            lineType=cv2.LINE_AA,
-                        )
+                all_points_lists = []
+                for i in range(self.repetitions):
+                    # # dividing the line into points
+                    points_count = random.randint(3, 10)  # dividing the line into points
+                    points = np.linspace(starting_point[0], ending_point[0], points_count)
+                    relative_start_point = starting_point[1] if i % 2 else ending_point[1]
+                    points = [[int(x), (starting_point[1] + random.randint(-offset, offset))] for x in points]
+                    points = smooth(points, 6)  # adding a smoothing effect in points using chaikin's algorithm
+                    all_points_lists.append(points)
+
+                # points_count = random.randint(3, 10)
+                # points = np.linspace(starting_point[0], ending_point[0], points_count)
+                # points_list = [[int(x), (starting_point[1] + random.randint(-offset, offset))] for x in points]
+                # # adding a smoothing effect in points using chaikin's algorithm
+                # points_list = smooth(points_list, 8)
+
+                for k in range(len(all_points_lists)):
+                    points_list = all_points_lists[k]
+                    for i in range(len(points_list) - 1):
+                        p1 = (int(points_list[i][0]), int(points_list[i][1]))
+                        if self.markup_type == "highlight":
+                            p2 = (int(points_list[i + 1][0]), int(points_list[i + 1][1] - h))
+                            # A filled rectangle
+                            markup_mask = cv2.rectangle(markup_mask, p1, p2, self.markup_color, -1)
+                        else:
+                            p2 = (int(points_list[i + 1][0]), int(points_list[i + 1][1]))
+                            markup_mask = cv2.line(
+                                markup_mask,
+                                p1,
+                                p2,
+                                self.markup_color,
+                                markup_thickness,
+                                lineType=cv2.LINE_AA,
+                            )
 
         # smoothen highlight mask to make it more realistic
-        markup_mask = cv2.GaussianBlur(markup_mask, (7, 7), cv2.BORDER_DEFAULT)
+        if self.markup_type == "highlight":
+            alpha = 0.5
```

Comment:
With `alpha` = 0.5, the highlighted colour intensity will be reduced, did you try with 0.5 and 1 and see the differences?

Reply:
> The whole purpose is to draw strikethrough or underline into the `markup_mask` first, then we may need to apply some filters on it before the overlay process. Otherwise, if we apply them directly to the image, those filters will affect the image too, which in some cases they shouldn't.

Actually, the overlay is still being applied but for strikethrough and underline the value of alpha is 1 because these markups are mostly drawn with solid colors.

Reply:
> > The whole purpose is to draw strikethrough or underline into the `markup_mask` first, then we may need to apply some filters on it before the overlay process. Otherwise, if we apply them directly to the image, those filters will affect the image too, which in some cases they shouldn't.
> 
> Actually, the overlay is still being applied but for strikethrough and underline the value of alpha is 1 because these markups are mostly drawn with solid colors.

Yea, actually i meant for highlight, why you set it 0.5? That will reduce intensity of highlighted colour since the alpha value is lower.